### ISSUE TITLE
Building new objects with Flexible extension 

### ISSUE THREAD

**Issue by chinnuGarg** [reinterpretcat/utymap#129]
Hello reinterpret cat, great library.

I am colliding Ethan 3D model to the building and trying to get latlong of the corners for the same building.
If Ethan 3D model has already collided with the building, I will add a 3D cube on top of that building (as in https://github.com/reinterpretcat/utymap#flexible-extension) indicating the building’s location is already found.

On Collision of Ethan to the building, I am just getting the id of the building nothing more than that. And we need a node to use flexible extension. 
So is it possible to do this?

**Comment by reinterpretcat** [comment]
You could try to add your collision detection script via https://github.com/reinterpretcat/utymap/blob/master/unity/demo/Assets/Scripts/Core/Plugins/GameObjectBuilder.cs::BuildFromMesh
Buildings are built in core library and they are materialized using BuildFromMesh method.

Another way is just to search for building's GameObject in scene using its id and knowing how are building names are generated.

Another way might involve search api implemented in core lib

**Comment by reinterpretcat** [comment]
Have you tried to store lat/lon in your custom MonoBehaviour when you're attaching it to gameobject?

**Comment by chinnuGarg** [comment]
What is the way to get the latlon for the buildings? Can I get it while i build the mesh in GameObjectBuilder css?

**Comment by reinterpretcat** [comment]
No, mesh geometry is already mapped to unity coordinates. Here you have raw geometry: https://github.com/reinterpretcat/utymap/blob/master/unity/demo/Assets/Scripts/Core/Interop/MapDataAdapter.cs

**Comment by chinnuGarg** [comment]
Is it possible to create custom shaped building rather than rectangular buildings with the current raw data that OSM provides??

**Comment by reinterpretcat** [comment]
You can change or add new implementation for that based on current building builder implementation: https://github.com/reinterpretcat/utymap/tree/master/core/src/builders/buildings

**Comment by chinnuGarg** [comment]
Does UtyMap provides any kind of latlong bounds to determine the centre of polygon or a building?
